shared/Microsoft.Extensions.StackTrace.Sources/ExceptionDetailsProvider.cs

```diff
@@ -57,11 +57,24 @@ private static IEnumerable<Exception> FlattenAndReverseExceptionTree(Exception e
             }
 
             var list = new List<Exception>();
-            while (ex != null)
+            if (ex.GetType().IsAssignableFrom(typeof(AggregateException)))
             {
                 list.Add(ex);
-                ex = ex.InnerException;
+                var exception = (AggregateException) ex;
+                foreach (var innerException in exception.Flatten().InnerExceptions)
+                {
+                    list.Add(innerException);
+                }
+            }
+            else
+            {
+                while (ex != null)
+                {
+                    list.Add(ex);
+                    ex = ex.InnerException;
+                }
             }
+
             list.Reverse();
```

Comment:
Can you move this inside the `else`? Since we're not trying to invert the exception callstack in the AggregateException case